crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/stateless/StatelessAssessmentService.java

```diff
@@ -43,88 +39,86 @@ public StatelessResult execute(Assessment assessment,
                                    Map<AgeRange, Integer> childGroupings,
                                    List<Income> income,
                                    List<Outgoing> outgoings) {
+        final var assessmentCriteriaEntity = assessmentCriteriaService.getAssessmentCriteria(
+                assessment.getAssessmentDate(), assessment.getHasPartner(), false
+        );
+        var initialAnswer = initialResult(
+                childGroupings,
+                assessmentCriteriaEntity,
+                assessment.getCaseType(),
+                assessment.getMagistrateCourtOutcome(),
+                income,
+                null);
 
         if (assessment.getAssessmentType() == StatelessRequestType.INITIAL) {
-            return initialOnly(assessment, childGroupings, income);
+            return new StatelessResult(null, initialAnswer);
         } else {
-            return initialAndFull(assessment, childGroupings, income, outgoings);
+            if (initialAnswer.isFullAssessmentPossible()) {
+                final var statelessFullResult =
+                        fullResult(childGroupings, assessmentCriteriaEntity, assessment.getEligibilityCheckRequired(), assessment.getCaseType(), outgoings,
+                                initialAnswer.getTotalAggregatedIncome());
+                return new StatelessResult(statelessFullResult, initialAnswer);
+            } else {
+                return new StatelessResult(null, initialAnswer);
+            }
         }
     }
 
-    private StatelessResult initialAndFull(@NotNull Assessment assessment,
-                                           @NotNull Map<AgeRange, Integer> childGroupings,
-                                           @NotNull List<Income> incomes,
-                                           @NotNull List<Outgoing> outgoings) {
+    public StatelessFullResult fullResult(Map<AgeRange, Integer> childGroupings,
```

Comment:
Should the `childGroupings` have the `@NotNull` annotation the same as the `initialResult` method does? I think it would still result in a `NullPointerException`.

Reply:
@starswan has done this too